### PR TITLE
Storybook: Add DocsPage and test out static content

### PR DESCRIPTION
**Pull request by lisasy**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## Description

In this PR, I'm doing a few things:
- [x] Adding the DocsPage add-on
- [x] Updating the static pages (content), and making them into `mdx` pages

## Related Tickets & Documents
Closes https://github.com/forem/InternalProjectPlanning/issues/281

## QA Instructions, Screenshots, Recordings
To test Storybook locally on your machine, run these commands:

```
yarn
yarn storybook
```
Once you run `yarn storybook`, the localhost should open up in your default browser. For me, it was `localhost:6006`.

### Proposed changes
Screen Shot 2020-11-16 at 1 18 40 PM

### Before
Screen Shot 2020-11-16 at 1 19 02 PM

## Added tests?

- [ ] Yes
- [x] No, and this is why: I'm not sure if we have tests for the updates that I'm suggesting.
- [ ] I need help with writing tests

## Added to documentation?

- [ ] [Developer Docs](https://docs.forem.com) and/or [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [ ] README
- [ ] No documentation needed

### CODE REVIEW

**Comment by nickytonline** [comment]
@lisasy, I've fixed the conflicts and updated the branch with the latest from the main branch;

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -40,7 +40,8 @@ as any relevant images for UI changes._
 
 ## Added to documentation?
 
-- [ ] [Developer Docs](https://docs.forem.com) and/or [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
+- [ ] [Developer Docs](https://docs.forem.com) and/or
```
Comment (nickytonline) [inline .github/PULL_REQUEST_TEMPLATE.md:43]: Looks like prettier ran when I merged master before committing.

File: README.md [hunk 2]
```diff
@@ -166,7 +166,8 @@ We welcome security research on DEV under the terms of our
 
 ## Acknowledgments
 
-Thank you to the [Twemoji project](https://github.com/twitter/twemoji) for the usage of their emojis.
+Thank you to the [Twemoji project](https://github.com/twitter/twemoji) for the
```
Comment (nickytonline) [inline README.md:169]: Looks like prettier ran when I merged master before committing.

File: app/assets/images/circle-question.svg [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="54px" height="54px">
-  <path fill="none" d="M0 0h24v24H0z"/>
-  <path fill="#ffffff" d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-2a8 8 0 1 0 0-16 8 8 0 0 0 0 16zm-1-5h2v2h-2v-2zm2-1.645V14h-2v-1.5a1 1 0 0 1 1-1 1.5 1.5 0 1 0-1.471-1.794l-1.962-.393A3.501 3.501 0 1 1 13 13.355z"/>
-</svg>
\ No newline at end of file
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="54" height="54">
```
Comment (nickytonline) [inline app/assets/images/circle-question.svg:1]: Looks like prettier ran when I merged master before committing.

File: app/assets/images/logo-forem.svg [hunk 4]
```diff
@@ -1 +1,10 @@
-<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg"><g clip-path="url(#clip0)" fill="#1AB3A6"><path d="M4.603 1.438a8.056 8.056 0 017.643 5.478 8.543 8.543 0 00-3.023 5.968H8.054C3.606 12.884 0 9.296 0 4.87V1.468a.03.03 0 01.03-.03h4.573zM23.97 6.515a.03.03 0 01.03.03v2.833c0 4.11-3.354 7.442-7.491 7.442h-2.881v5.726h-2.305V14.53l.022-1.145c.294-3.843 3.526-6.87 7.469-6.87h5.155z"/></g><defs><clipPath id="clip0"><path fill="#fff" d="M0 0h24v24H0z"/></clipPath></defs></svg>
+<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (nickytonline) [inline app/assets/images/logo-forem.svg:1]: Looks like prettier ran when I merged master before committing.

File: app/assets/javascripts/initializers/initializeEllipsisMenu.js [hunk 5]
```diff
@@ -47,9 +47,7 @@ function onXhrSuccess(form, article, values) {
     toggleNotifications(submit, submitValue);
   }
 
-  article
-    .getElementsByClassName('js-ellipsis-menu')[0]
-    .classList.add('hidden');
+  article.getElementsByClassName('js-ellipsis-menu')[0].classList.add('hidden');
```
Comment (nickytonline) [inline app/assets/javascripts/initializers/initializeEllipsisMenu.js:50]: Looks like prettier ran when I merged master before committing.

File: app/assets/stylesheets/components/modals.scss [hunk 6]
```diff
@@ -76,7 +76,7 @@
       justify-content: space-between;
       padding: var(--modal-header-padding);
       flex-shrink: 0;
-      
+
```
Comment (nickytonline) [inline app/assets/stylesheets/components/modals.scss:79]: Looks like prettier ran when I merged master before committing.

File: app/assets/images/circle-question.svg (hunk 3 above)
Comment (lisasy) [reply app/assets/images/circle-question.svg:1]: If Prettier ran, does this mean we want to roll this change back?

**Comment by ludwiczakpawel** [comment]
Can you give instructions on how to test things?

**Comment by lisasy** [comment]
> Can you give instructions on how to test things?

@ludwiczakpawel To test Storybook locally on your machine, run these commands:

```
yarn
yarn storybook
```
Once you run `yarn storybook`, the localhost should open up in your default browser. For me, it was `localhost:6006`.

Let me know if there's anything more you need!

**Comment by nickytonline** [comment]
When you run `yarn storybook`, it should open the browser automatically at the address @lisasy mentioned.

**Comment by ludwiczakpawel** [comment]
oh lol, yea. i forgot how to run storybook :D

File: app/assets/stylesheets/config/_import.scss [hunk 7]
```diff
@@ -2,7 +2,7 @@
 // Colors. /////////////////////////////////////////////////////////////////////////
 ////////////////////////////////////////////////////////////////////////////////////
 //
-// List of colors to be used to generate utility-first classes,
+// List of colors to be used to generate Utility-First Classes/,
```
Comment (lisasy) [inline app/assets/stylesheets/config/_import.scss:5]: Undo this change

**Comment by lisasy** [comment]
> oh lol, yea. i forgot how to run storybook :D
> 
> But what exactly should I test here? Static pages (Introduction, Component Library, Utility-First CSS, Writing CSS) seem to look ok. But when I go to other places and then preview things in the "Docs" tab, the preview panels take a lot of time to load, like several seconds (not sure why?).
> And when I choose "show code" option, I don't think the output is what we expect it to be, right?

All this PR is about is the addition of the DocsPage plug-in and the static pages. When you preview things in the "Docs" tab, it isn't working well right now, and we'll need to open small PRs to begin updating everything so that it works well :P 

For "Show code", that's also a detail we'll need to configure in another PR! 

If the static pages look okay, we can move forward with this PR :) Keeping it small.

**Comment by lisasy** [comment]
@nickytonline I'm running into file conflicts again with `package.json` and `yarn.lock` ... Can you help me out here?

**Comment by nickytonline** [comment]
> @nickytonline I'm running into file conflicts again with `package.json` and `yarn.lock` ... Can you help me out here?

Taking a peek and also giving it a review.

**Comment by nickytonline** [comment]
@lisasy, grab latest of your branch. I've updated it with the latest from the main branch and fixed the conflicts.

**Review by nickytonline — APPROVED** [review]
I tested the static pages and agree we can work on the other stuff in upcoming PRs. Thanks for the PR Lisa!

**Comment by nickytonline** [comment]
I put up PR #11586 to help with the extra files showing up in your PR from prettier formatting.

File: app/assets/images/circle-question.svg (hunk 3 above)
Comment (nickytonline) [reply app/assets/images/circle-question.svg:1]: Actually prettier doesn't run on svg, so not sure what happened here.

**Comment by lisasy** [comment]
@nickytonline Yikes! Does the Travis CI errors indicate that I installed the add-on plugin incorrectly (e.g. some syntax issue)?

**Comment by nickytonline** [comment]
> @nickytonline Yikes! Does the Travis CI errors indicate that I installed the add-on plugin incorrectly (e.g. some syntax issue)?

Let me take a peek.

**Comment by nickytonline** [comment]
@lisasy, we have some flaky tests. One of the build nodes failed, but the other two passed. I restarted the failing one and then it should be good.

**Comment by ludwiczakpawel** [comment]
build still failing - not sure if it's flaky test though or something actually related to the storybook..? cc @nickytonline

**Comment by lisasy** [comment]
@nickytonline It's seeming like the build is failing a lot.... is there a way we can address this, have the build succeed, and merge this into `main` before this issue becomes too stale?

**Comment by nickytonline** [comment]
@lisasy, I've restarted the build node that was failing. It's a flaky test causing the issue.

**Comment by nickytonline** [comment]
> @nickytonline It's seeming like the build is failing a lot.... is there a way we can address this, have the build succeed, and merge this into `main` before this issue becomes too stale?

There are known flaky tests that will be addressed at some point. Typically one out of the three build nodes fail, at least in my experience. I believe @mstruve and @maestromac are aware of this.

**Comment by nickytonline** [comment]
@lisasy, I thought it was a flaky test, but it's an actual error.

```bash
ERR! ModuleNotFoundError: Module not found: Error: Can't resolve '@storybook/addon-docs/blocks' in '/home/travis/build/forem/forem/app/javascript/.storybook'
```

File: package.json [hunk 8]
```diff
@@ -87,6 +87,7 @@
     "@stoplight/spectral": "5.7.1",
     "@storybook/addon-a11y": "^6.1.5",
     "@storybook/addon-actions": "^6.1.5",
+    "@storybook/addon-docs": "^6.1.9",
```
Comment (lisasy) [inline package.json:90]: The culprit!

**Review by ludwiczakpawel — APPROVED** [review]

File: app/assets/images/circle-question.svg (hunk 3 above)
Comment (ludwiczakpawel) [reply app/assets/images/circle-question.svg:1]: probably SVGO.

**Comment by lisasy** [comment]
@nickytonline Do you have any ideas on why the CodeQL is failing and how we can resolve it? 

Its error message is really nondescript 🥴

Screen Shot 2020-12-02 at 9 11 22 AM

**Comment by nickytonline** [comment]
> @nickytonline Do you have any ideas on why the CodeQL is failing and how we can resolve it?
> 
> Its error message is really nondescript 🥴
> 
> Screen Shot 2020-12-02 at 9 11 22 AM

@lisasy, it seems like the job didn't run. I restarted it.
